### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: taxi/forms.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    def clean_license_number(self):
+        license_number = self.cleaned_data["license_number"]
+        if len(license_number) != 8:
+            raise ValidationError(
+                f"Consist only of {DriverLicenseUpdateForm.LICENSE_LEN}"
+                "characters"
+            )
+        if (
+            not license_number[
+                : DriverLicenseUpdateForm.INT_START_INDEX
```
Comment (ArsenPidhoretskyi) [inline taxi/forms.py:22]:
```suggestion
                : self.INT_START_INDEX
```
You can use self. Fix in all places

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: taxi/forms.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    LICENSE_LEN = 8
+    INT_START_INDEX = 3
+
+    class Meta:
+        model = Driver
+        fields = ("license_number",)
+
+    def clean_license_number(self):
+        license_number = self.cleaned_data["license_number"]
+        if len(license_number) != 8:
```
Comment (ArsenPidhoretskyi) [inline taxi/forms.py:15]: 8 -> self.LICENSE_LEN

File: taxi/forms.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            or not license_number[: self.INT_START_INDEX].isalpha()
+        ):
+            raise ValidationError(
+                f"First {self.INT_START_INDEX}"
+                " characters are uppercase letters"
+            )
+        if not license_number[self.INT_START_INDEX :].isdigit():
+            must_be_letters = self.LICENSE_LEN - self.INT_START_INDEX
+            raise ValidationError(
+                f"First {must_be_letters} characters are uppercase letters"
```
Comment (ArsenPidhoretskyi) [inline taxi/forms.py:30]:
```suggestion
                f"Last {must_be_letters} characters are uppercase letters"
```

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: templates/taxi/driver_form.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+{% block content %}
+  <h1>{{ object|yesno:"Update,Create" }} driver</h1>
+  <form action="" method="post" novalidate>
+    {% csrf_token %}
+    {{ form|crispy }}
+
+    <input type="submit" value="Submit" class="btn btn-primary">
+  </form>
+{% endblock %}
```
Comment (vsmutok) [inline templates/taxi/driver_form.html:12]: add new empty line

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Review by vasylhnatiuk — APPROVED** [review]

**Pull request closed without merging**